_posts/2023-01-04-trino-2022-the-rabbit-reflects.md

```diff
@@ -0,0 +1,579 @@
+ data types from and to Parquet in general. There were also general performance
+ to ORC types, and now have the ability to write Bloom filters in ORC files. We
+ have also improved performance and efficiency for a wide range of ORC and
+ Parquet-related operations.
+
+These improvements in aggregate are at the core of what makes Trino fast. There
+is no silver bullet you can plug in to speed things up. It takes time, effort,
+and smart changes to improve the speed of various systems.
+
+### Runtime improvements
+
+[Trino upgraded Java versions](https://github.com/trinodb/trino/issues/9876)from
+Java 11 to the next long-term support version of Java, which was Java 17. This
+upgrade improves the latency, memory footprint, startup, ramp up, pause times,
+and shut down times for Trino. Along with these performance enhancements one of the other
+critical elements of improving performance for the Java Virtual Machine is to
+improve the garbage collector. Trino uses the G1 garbage collector which can now
+more efficiently reclaim memory and reduce pause times.
+
+Martin did a benchmark on Trino workloads, and reported seeing anywhere from
+10-15% improvement in speed in Java 17. Many users have also reported a similar
+boost in their workloads, and we have yet to have a user mention any regressions
+related to Java 17, which is exciting news. Aside from having to perform the
+upgrades, we get a lot of these performance enhancements for free.
+
+On top of performance, upgrading to Java 17 adds new Java language features to
+improve the ability to write and maintain higher quality code. Which matters a
+lot in a fast-moving open-source project like Trino. Java 17 also adds a Vector
+API, which allows Trino to explore running some algorithms on native hardware,
+ideally resulting in a speed-up from Java's exceptional auto-vectorization.
+
+To learn more, read [this blog 
+post]({% post_url 2022-07-14-trino-updates-to-java-17 %}) and watch this episode
+of [the Trino Community Broadcast]({{site.url}}/episodes/36.html)
+
+Along with the Java upgrade, Trino now has a Docker image for ppc64le and added
+CLI support for ARM64, which means Trino's Docker image and the CLI can run on
+the new MacBooks.
+  
+### Security
+
+Trino added the following improvements and features relevant for authentication,
+authorization and integration with other security systems:
+
+* There were a lot of updates to [OAuth 2.0
+ authentication]({{site.url}}/docs/current/security/oauth2.html) like support for OAuth
+ 2.0 refresh tokens and allowing access token passthrough with refresh tokens
+ enabled. We also added support for [automatic discovery of OpenID
+ Connect]({{site.url}}/docs/current/security/oauth2.html#openid-connect-discovery)
+ metadata with OAuth 2.0 authentication. Support for groups in OAuth2 claims and
+ reduced latency for OAuth2.0 authentication.
+* Access control rules got a fair amount of attention. Support for string
```

Comment:
I feel like this whole item could be dropped

Reply:
dropped